### PR TITLE
New: no-negated-condition rule (fixes #3740)

### LINKED ISSUE #3740: Proposal: no-negated-condition
I'd like to forbid:
```js
if (!a) {
    console.log(1);
} else {
    console.log(2);
}
```
in favor of:
```js
if (a) {
    console.log(2);
} else {
    console.log(1);
}
```
JSCS has a rule of that effect named [disallowNotOperatorsInConditionals](http://jscs.info/rule/disallowNotOperatorsInConditionals).

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26603300-proposal-no-negated-condition?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by alberto**

### CODE REVIEW

File: docs/rules/no-negated-condition.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Disallow use of negated expressions in conditions (no-negated-condition)
+
+Checks against the use of a negated expression in an if condition when the else branch is empty or in a ternary operator.
```
Comment (ilyavolodin) [inline docs/rules/no-negated-condition.md]: This description seems to not be inline with provided examples and code of the rule. It looks like if else branch is empty you exit the rule.

Comment (alberto) [inline docs/rules/no-negated-condition.md]: @ilyavolodin You are right. What about this? "Checks against the use of a negated expression in a ternary operator or in an if condition when the else branch is not empty."

Comment (ilyavolodin) [inline docs/rules/no-negated-condition.md]: @alberto That sounds much better I think.

File: lib/rules/no-negated-condition.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                return;
+            }
+            if ((node.test.type === "UnaryExpression" && node.test.operator === "!") ||
+                (node.test.type === "BinaryExpression" &&
+                (node.test.operator === "!=" || node.test.operator === "!=="))) {
+                context.report(node, "Unexpected negated condition.");
+            }
+        },
+        "ConditionalExpression": function(node) {
+            if (node.test.type === "UnaryExpression" && node.test.operator === "!") {
```
Comment (michaelficarra) [inline lib/rules/no-negated-condition.js]: You should probably also have the BinaryExpression test here, to align with IfStatement test handling.

File: tests/lib/rules/no-negated-condition.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester();
+ruleTester.run("no-negated-condition", rule, {
+
+    // Examples of code that should not trigger the rule
+    valid: [
+        "if (a) {;}",
```
Comment (michaelficarra) [inline tests/lib/rules/no-negated-condition.js]: These tests with `{;}` are weird. Prefer `{}` or `;`.

**Comment by alberto** [comment]
Any feedback on this? I updated the code with the recommendations from above?

File: docs/rules/no-negated-condition.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+# Disallow use of negated expressions in conditions (no-negated-condition)
+
+Checks against the use of a negated expression in an if condition when the else branch is empty or in a ternary operator. Negated conditions are more difficult to understand. Code can be made more readable by inverting the condition instead.
```
Comment (ilyavolodin) [inline docs/rules/no-negated-condition.md]: Wait, I though this was fixed already. This rule checks if condition when the else branch is **not** empty.

**Comment by ilyavolodin** [comment]
Sorry, lost track of this PR. Looks fine, with exception of documentation change noted above.

File: docs/rules/no-negated-condition.md (hunk 4 above)
Comment (alberto) [inline docs/rules/no-negated-condition.md]: Ooops, I somehow forgot to include it in the commit. I'll fix it.

File: lib/rules/no-negated-condition.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
+/**
+ * @fileoverview Rule to disallow a negated condition
+ * @author Brandon Mills
```
Comment (nzakas) [inline lib/rules/no-negated-condition.js]: Please add your name and copyright

File: tests/lib/rules/no-negated-condition.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
+/**
+ * @fileoverview Tests for no-negated-condition rule.
+ * @author Brandon Mills
```
Comment (nzakas) [inline tests/lib/rules/no-negated-condition.js]: Please add your name and copyright

**Comment by alberto** [comment]
Added name and copyright

**Pull request merged**